### PR TITLE
feat(backend): aggregate worker

### PR DESCRIPTION
**Pull request by aldy505**
Closes https://github.com/teknologi-umum/semyi/issues/24

### CODE REVIEW

File: backend/aggregate_worker.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+				continue
+			}
+
+			// Filter out for today's data
+			var now = time.Now()
+			var fromTime = time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location())
+			var toTime = fromTime.Add(24 * time.Hour)
+			var lastHourData []MonitorHistorical
+			for _, data := range historicalData {
+				if data.Timestamp.After(fromTime) && data.Timestamp.Before(toTime) {
```
Comment (YogiPristiawan) [inline backend/aggregate_worker.go:85]: Should the upper and lower limits be included? Maybe something like
```suggestion
				if (data.Timestamp.After(fromTime) || data.Timestamp.Equal(fromTime)) && (data.Timestamp.Before(toTime) || data.Timestamp.Equal(toTime)) {
```

File: backend/aggregate_worker.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			// Filter out from the last hour
+			// If right now is 08:29, we should filter out data from 08:00 - 09:00
+			// If right now is 08:01, we should filter out data from 08:00 - 09:00
+			// If right now is 09:00, we should filter out data from 09:00 - 10:00
+			var now = time.Now()
+			var fromTime = time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), 0, 0, 0, now.Location())
+			var toTime = fromTime.Add(1 * time.Hour)
+			var lastHourData []MonitorHistorical
+			for _, data := range historicalData {
+				if data.Timestamp.After(fromTime) && data.Timestamp.Before(toTime) {
```
[the remark this answers is not in the document]
Comment (aldy505) [reply backend/aggregate_worker.go:40]: I don't think this is needed though, but if we want to improve it, we might just do this:

```suggestion
				if data.Timestamp.Equal(fromTime) || (data.Timestamp.After(fromTime) && data.Timestamp.Before(toTime)) {
```

File: backend/aggregate_worker.go (hunk 1 above)
Comment (aldy505) [reply backend/aggregate_worker.go:85]: Similar to my previous note.

```suggestion
				if data.Timestamp.Equal(fromTime) || (data.Timestamp.After(fromTime) && data.Timestamp.Before(toTime)) {
```

**Review by YogiPristiawan — APPROVED** [review]

**Review by YogiPristiawan — APPROVED** [review]
leres

**Pull request merged**